### DISCUSSION TITLE
How can i test custom hooks like this?

### DISCUSSION THREAD

**Discussion by liujiapeng** [vuejs/test-utils#1229] [Q&A] [answered]
I read some tutorials for testing the composition api, but it looks like they all use the html output to test correctness. Sometimes I use custom hooks to write some logic, but I don't know how to test it, is there a demo for it?
```js
import { ref } from 'vue'
const useService = () => {
   const count = ref(0)
   const addCount = () => count.value++
   
   return {
     count,
     addCount
   }
}
```

**Comment by cexbrayat** [comment, marked as answer]
Hi @liujiapeng 

You don't need Vue Test Utils to test this: it's a simple function test.

I would write something like this (I did not test it):

```ts
test('addCount should increment count', () => {
  const service = useService();
  expect(service.count.value).toBe(0);
  service.addCoun();
  expect(service.count.value).toBe(1);
});
```

**Reply by liujiapeng** [reply]
it really works, thank u so much!!

**Reply by whitespacecode** [reply]
@cexbrayat
So you don't need to mount the component to test the useService ?

**Reply by cexbrayat** [reply]
@whitespacecode In most cases, you don't need to mount a component to test a composable.
